include/easi/component/OptimalStress.h

```diff
@@ -99,7 +100,16 @@ void easi::OptimalStress::evaluate() {
   o.b_yy = -((ci * ci * s11 * sr * sr + s33 * si * si) * cd * cd + 2 * ci * sd * si * sr * (s33 - s11) * cd + (s33 * sd * sd * sr * sr + cr * cr * s22) * ci * ci + sd * sd * si * si * s11) * ss * ss - 2 * cr * cs * (cd * cd * ci * sr * s11 + sd * si * (s33 - s11) * cd - ci * sr * (-s33 * sd * sd + s22)) * ss - cs * cs * (cd * cd * cr * cr * s11 + cr * cr * s33 * sd * sd + s22 * sr * sr);
   o.b_yz = -cd * sd * sr * ss * (s33 - s11) * si * si - ((-ss * (-s11 * sr * sr + s33) * ci + cr * cs * sr * s11) * cd * cd + ss * ((s33 * sr * sr - s11) * sd * sd + cr * cr * s22) * ci - cr * cs * sr * (-s33 * sd * sd + s22)) * si + cd * ci * sd * (ci * sr * ss + cr * cs) * (s33 - s11);
   o.b_zz = -(cd * cd * s11 * sr * sr + s33 * sd * sd * sr * sr + cr * cr * s22) * si * si - 2 * cd * ci * sd * sr * (s11 - s33) * si - ci * ci * (cd * cd * s33 + s11 * sd * sd);
-
+  
+  if (abs(i.s_zz)>0.0) {
+    double factor = -effectiveConfiningStress/o.b_zz;
+    o.b_xx = o.b_xx * factor;  
+    o.b_yy = o.b_yy * factor;  
+    o.b_zz = o.b_zz * factor;  
+    o.b_xy = o.b_xy * factor;  
+    o.b_xz = o.b_xz * factor;  
+    o.b_yz = o.b_yz * factor;  
+  }
```

Comment:
Why do you introduce s_zz as new parameter? The same functionality could be achieved by adding an AffineMap or FunctionMap?

Reply:
My point is that it makes more sense in most cases (all?) to constrain sig_zz than Tr(s_ii)/3 to be equal to the effective weight of the overlying column.


An extreme case when that makes a difference is for a stress setup favoring normal and reverse faulting:
![image](https://user-images.githubusercontent.com/13536910/124496919-626e8d80-ddba-11eb-8e0e-aa3463635a64.png)
Mohr–Coulomb diagram at 5 km depth for normal and reverse faults illustrated in blue and red, respectively.
Source: https://doi-org.emedien.ub.uni-muenchen.de/10.1093/gji/ggy407

normal faulting: s1 = s_zz
reverse faulting s3 = s_zz.
Which is very different from what you get by constraining Tr(s_ii)/3.

So yes, we can write a FunctionMap to scale the value of sig_zz.
But this won't be convenient and understandable for the non-advanced easi user.
I m not very satisfied with this loss of backward compatibility, but it seems like the least worse solution.
Maybe we could go even further and remove 'effectiveConfiningStress' from the parameter list.

Reply:
The question is do we want to write that every time we use the module:
```    
    !EvalModel
    parameters: [b_xx,b_yy,b_zz,b_xy, b_xz, b_yz]
    model: !ConstantMap
        map:
                mu_d: a
        components: !OptimalStress
              constants:
                mu_s:      b
                strike:    c
                dip:       d
                rake:     90.0
                cohesion:  e
                s2ratio:   f
                R:         g
                effectiveConfiningStress: h
    components: !FunctionMap
        map:
           b_xx: return b_xx* (b_xx+b_yy+b_zz)/(3 *b_zz);
           b_yy: return b_yy* (b_xx+b_yy+b_zz)/(3 *b_zz);
           b_zz: return (b_xx+b_yy+b_zz)/3;
           b_xy: return b_xy* (b_xx+b_yy+b_zz)/(3 *b_zz);
           b_xz: return b_xz *(b_xx+b_yy+b_zz)/(3 *b_zz);
           b_yz: return b_yz* (b_xx+b_yy+b_zz)/(3 *b_zz);
           ```

Reply:
Sorry if I don't get it, but say you only specify only s_zz (and drop effectiveConfiningStress). Could you recover the behaviour of effectiveConfiningStress with a FunctionMap? If yes, then I'd make s_zz the default and drop effectiveConfiningStress.